pkg/apis/crd/v1alpha1/types.go

```diff
@@ -430,8 +430,13 @@ type Rule struct {
 	Name string `json:"name,omitempty"`
 	// EnableLogging is used to indicate if agent should generate logs
 	// when rules are matched. Should be default to false.
+	// This field can't be used with LogSettings, will be deprecated.
 	// +optional
 	EnableLogging bool `json:"enableLogging"`
+	// LogSettings is used to specify logging behaviors of the rules when matched.
+	// This field can't be used with EnableLogging.
+	// +optional
+	LogSettings *RuleLogSettings `json:"logSetting,omitempty"`
```

Comment:
I wonder if we really need to make a new struct and deprecate the old field, considering:
1. Could we export the value to flow record for visbility as well, which adds one dimension that can be used to aggregate flows? Then the field doesn't need to be specific to log, and doesn't really need logging to be enabled.
2. Similar API in products like NSX doesn't make the field specific to logging as well.

Perhaps adding a new field of string could fullfil the requirement and be more extensible when we want to leverage the value in other features like flow visibility, and can avoid the extra effort to deprecate the old field:
```
// Label is user-defined arbitrary string which will be printed in the NetworkPolicy
// logs and exported in flow records. 
Label string `json:"label,omitempty"`
```

cc @jianjuns @antoninbas @salv-orlando @qiyueyao

Reply:
+1 to not making it specific to audit logs

Reply:
Sure that makes sense, didn't consider these use cases. Updating the CRD.